kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/admin/ProcessAdminServiceBase.java

```diff
@@ -90,6 +99,35 @@ public MigrationReportInstanceList migrateProcessInstances(String containerId, L
         return convertMigrationReports(reports);
     }
 
+    public MigrationReportInstanceList migrateProcessInstanceWithAllSubprocess(String containerId,
+                                                                               Long processInstanceId,
+                                                                               String targetContainerId,
+                                                                               String targetProcessId,
+                                                                               String payload, 
+                                                                               String marshallingType) {
+
+        ProcessInstanceDesc pi = runtimeDataService.getProcessInstanceById(processInstanceId);
+        if(pi.getParentId() > 0) {
```

Comment:
parentId might be < 0 (for backward compatibility, there might be cases where the id is negative, if you remember about the negatives id generated). I think it would be safer if we check as below, wdyt?
```suggestion
        if(pi.getParentId() != 0) {
```

Reply:
negative number should not be there.